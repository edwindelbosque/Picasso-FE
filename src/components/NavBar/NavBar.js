import React, { useState, useEffect } from 'react';
import './NavBar.scss';
import Fade from 'react-reveal/Fade';
// import { createUser } from '../../util/apiCalls';
import Palettes from '../Palettes/Palettes';
import Catalogs from '../Catalogs/Catalogs';
import { Route, Link } from 'react-router-dom';

let catalogs = [
	{
		id: 1,
		catalogName: 'Personal',
		user_id: 1
	},
	{
		id: 2,
		catalogName: 'Favorites',
		user_id: 1
	},
	{
		id: 3,
		catalogName: 'Mood',
		user_id: 2
	},
	{
		id: 4,
		catalogName: 'Work',
		user_id: 3
	},
	{
		id: 5,
		catalogName: 'Favorites',
		user_id: 4
	},
	{
		id: 6,
		catalogName: 'Mood',
		user_id: 4
	},
	{
		id: 7,
		catalogName: 'Work',
		user_id: 4
	},
	{
		id: 8,
		catalogName: 'Favorites',
		user_id: 5
	},
	{
		id: 9,
		catalogName: 'Mood',
		user_id: 5
	},
	{
		id: 10,
		catalogName: 'Work',
		user_id: 5
	}
];

let palettes = [
	{
		id: 1,
		paletteName: 'Personal',
		color1: '334534',
		color2: '443224',
		color3: '345534',
		color4: '342534',
		color5: '342533',
		catalog_id: 1
	},
	{
		id: 2,
		paletteName: 'Sunny',
		color1: '342537',
		color2: '742532',
		color3: '142535',
		color4: '544344',
		color5: '342531',
		catalog_id: 3
	},
	{
		id: 3,
		paletteName: 'I love it',
		color1: '542531',
		color2: '342537',
		color3: '342531',
		color4: '242536',
		color5: '742533',
		catalog_id: 2
	},
	{
		id: 4,
		paletteName: '12345',
		color1: '242531',
		color2: '342533',
		color3: '642536',
		color4: '142537',
		color5: '642530',
		catalog_id: 1
	},
	{
		id: 5,
		paletteName: 'Personal',
		color1: '042534',
		color2: '342534',
		color3: '042534',
		color4: '342534',
		color5: '042534',
		catalog_id: 4
	},
	{
		id: 6,
		paletteName: 'Dreamin',
		color1: '142534',
		color2: '342534',
		color3: '142534',
		color4: '342534',
		color5: '142534',
		catalog_id: 5
	},
	{
		id: 7,
		paletteName: 'Hotel',
		color1: '642534',
		color2: '642534',
		color3: '342534',
		color4: '642534',
		color5: '342534',
		catalog_id: 6
	},
	{
		id: 8,
		paletteName: 'Blue',
		color1: '842534',
		color2: '142534',
		color3: '342534',
		color4: '142534',
		color5: '342534',
		catalog_id: 7
	},
	{
		id: 9,
		paletteName: 'Warm',
		color1: '342534',
		color2: '742534',
		color3: '642534',
		color4: '842534',
		color5: '142534',
		catalog_id: 7
	},
	{
		id: 10,
		paletteName: 'Sad',
		color1: '142534',
		color2: '342534',
		color3: '142534',
		color4: '342534',
		color5: '142534',
		catalog_id: 8
	},
	{
		id: 11,
		paletteName: 'Colorful',
		color1: '142534',
		color2: '342534',
		color3: '142534',
		color4: '342534',
		color5: '142534',
		catalog_id: 9
	},
	{
		id: 12,
		paletteName: 'Vibrant',
		color1: '742534',
		color2: '342534',
		color3: '842534',
		color4: '342534',
		color5: '342534',
		catalog_id: 10
	},
	{
		id: 13,
		paletteName: 'Bold',
		color1: '942534',
		color2: '342534',
		color3: '142534',
		color4: '242534',
		color5: '342534',
		catalog_id: 10
	},
	{
		id: 14,
		paletteName: 'Magical',
		color1: '142534',
		color2: '742534',
		color3: '342534',
		color4: '142534',
		color5: '342534',
		catalog_id: 3
	},
	{
		id: 15,
		paletteName: 'Interesting',
		color1: '942534',
		color2: '042534',
		color3: '342534',
		color4: '142534',
		color5: '342534',
		catalog_id: 5
	},
	{
		id: 16,
		paletteName: 'Prototype',
		color1: '442534',
		color2: '342534',
		color3: '742534',
		color4: '142534',
		color5: '742534',
		catalog_id: 10
	},
	{
		id: 17,
		paletteName: 'Prototype Hotel 2',
		color1: '142534',
		color2: '842534',
		color3: '242534',
		color4: '342534',
		color5: '842534',
		catalog_id: 10
	},
	{
		id: 18,
		paletteName: 'Prototype Hotel 3',
		color1: '142534',
		color2: '342534',
		color3: '142534',
		color4: '142534',
		color5: '842534',
		catalog_id: 10
	},
	{
		id: 19,
		paletteName: 'Prototype Hotel 4',
		color1: '142534',
		color2: '742534',
		color3: '242534',
		color4: '342534',
		color5: '742534',
		catalog_id: 10
	},
	{
		id: 20,
		paletteName: 'Movie poster 1',
		color1: '142534',
		color2: '842534',
		color3: '642534',
		color4: '342534',
		color5: '242534',
		catalog_id: 3
	}
];

const NavBar = () => {
	const [menuIsActive, toggleMenu] = useState(false);
	const isSignedIn = true;

	useEffect(() => {
		// const postNewUser = async () => {
		// 	const newUser = {
		// 		email: 'edwindbdjba@gmail.com',
		// 		password: 'verySecret',
		// 		firstName: 'Edwin',
		// 		lastName: 'Del Bisquit'
		// 	};
		// 	const user = await createUser(newUser);
		// 	console.log(user);
		// };
		// postNewUser();
	}, []);

	return (
		<>
			<nav className='NavBar'>
				<p className={menuIsActive ? 'active-title' : ''}>Picasso</p>
				<Link
					to={
						isSignedIn
							? menuIsActive
								? '/create'
								: '/catalogs'
							: menuIsActive
							? '/create'
							: '/new-account'
					}>
					<div
						className={`hamburger-menu ${menuIsActive &&
							'hamburger-menu-active'}`}
						onClick={() => toggleMenu(!menuIsActive)}>
						<div className='bar-1'></div>
						<div className='bar-2'></div>
						<div className='bar-3'></div>
					</div>
				</Link>
			</nav>
			<div className={`menu ${menuIsActive && 'show-menu'}`}></div>
			<div className={`menu ${menuIsActive && 'show-menu'}`}>
				<Route path='/new-account'>
					<div className='access-buttons'>
						<Fade right when={menuIsActive} duration={300} delay={200}>
							<button
								className={`login-button ${menuIsActive && 'animate-button'}`}>
								Login
							</button>
							<button
								className={`signup-button ${menuIsActive && 'animate-button'}`}>
								Sign Up
							</button>
						</Fade>
					</div>
					<div className='information-area'>
						<Fade left when={menuIsActive} delay={60} duration={400}>
							<h1>
								Save all of your palettes by creating an <span>account.</span>
							</h1>
						</Fade>
						<div
							className={`hidden-circle ${menuIsActive &&
								'active-circle'}`}></div>
					</div>
				</Route>
				<div className='main-menu'>
					<Route path='/catalogs'>
						<Catalogs menuIsActive={menuIsActive} catalogs={catalogs} />
					</Route>
					<Route
						path='/catalogs/:id'
						render={({ match }) => {
							const matchingPalettes = palettes.filter(
								palette => palette.catalog_id === parseInt(match.params.id)
							);
							return (
								<Palettes
									menuIsActive={menuIsActive}
									palettes={matchingPalettes}
								/>
							);
						}}
					/>
				</div>
			</div>
		</>
	);
};

export default NavBar;
