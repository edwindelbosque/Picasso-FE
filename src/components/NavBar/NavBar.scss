$accent: #775c54;
$accent2: #2164f3;
$text-color: #000000;
$main: #ffffff;

.NavBar {
	height: 60px;
	position: fixed;
	padding: 0 20px;
	left: 0;
	top: 0;
	width: 100vw;
	display: flex;
	justify-content: space-between;
	align-items: center;
	z-index: 1;
	box-shadow: 0 2px 15px rgba(0, 0, 0, 0.13);
	backdrop-filter: blur(15px);
	background: rgba(255, 255, 255, 0.144);

	p {
		font-weight: 500;
		font-size: 25px;
		color: $accent2;
		font-weight: 600;
		letter-spacing: 0.5px;
		transition: 0.2s ease-in-out;
		cursor: default;
	}

	a {
		text-decoration: none;
	}

	h3 {
		margin-right: 57px;
		color: rgb(43, 43, 43);
		font-size: 16px;
		font-weight: 500;
		font-family: 'Roboto';
		cursor: default;
	}

	.active-title {
		color: $main;
	}

	.hamburger-menu {
		display: flex;
		position: fixed;
		right: 0;
		top: 0;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		cursor: pointer;
		width: 60px;
		height: 60px;
		transition: 0.16s;
		-webkit-tap-highlight-color: transparent;

		&:hover {
			background: $accent2;
			div {
				background: $main;
			}
		}

		div {
			height: 3px;
			width: 20px;
			background: $accent2;
			margin: 1.5px;
			transition: 0.16s;
		}
	}

	.hamburger-menu-active {
		align-items: center;
		background: $text-color;

		.bar-1 {
			height: 3px;
			width: 25px;
			background: $main;
			margin: 2px;
			position: relative;
			top: 6.5px;
			transform: rotate(-45deg);
		}

		.bar-2 {
			width: 0;
		}

		.bar-3 {
			height: 3px;
			transform: rotate(45deg);
			width: 25px;
			background: $main;
			position: relative;
			bottom: 6.5px;
			margin: 2px;
		}

		&:hover {
			background: $main;
			div {
				background: $accent2;
			}
		}
	}
}

.menu {
	background: $accent2;
	position: fixed;
	top: 100vh;
	left: 0;
	height: 100vh;
	width: 100vw;
	transition: 0.3s ease-in-out;

	.access-buttons {
		width: 100%;
		margin-top: 75px;
		height: 60px;
		display: flex;
		justify-content: flex-end;
	}

	button {
		height: 45px;
		width: 100px;
		cursor: pointer;
		margin-right: 15px;
		background: none;
		border: 1px solid $main;
		color: $main;
		border-radius: 3px;
		font-size: 18px;
		transition: 0.1s;

		&:focus {
			outline: none;
		}
	}

	.signup-button {
		background: $main;
		color: $accent;
		font-weight: 700;

		&:hover {
			background: $accent2;
			color: $main;
			border-color: $main;
		}
	}

	.login-button {
		transition: 0.1s;
		border: none;
		margin-right: 0;

		&:hover {
			color: rgb(216, 216, 216);
			transition: 0.1s;
		}
	}

	.information-area {
		height: 80%;
		width: 850px;
		padding-left: 80px;
		display: block;

		h1 {
			color: $main;
			font-size: 80px;
			font-family: 'Roboto';
			letter-spacing: -0.5px;

			span {
				border-bottom: 5px solid $accent2;
			}
		}
	}
}

.hidden-circle {
	height: 1200px;
	width: 1200px;
	background: $accent;
	position: relative;
	top: -950px;
	left: -300px;
	z-index: -1;
	border-radius: 50%;
	opacity: 0;
	transition: 0.2s;
	transition-delay: 0.2s;
}

.active-circle {
	opacity: 1;
}

.show-menu {
	top: 0px;
	transition: 0.3s ease-in-out;
}

.main-menu {
	width: 100%;
	background: $accent2;
	padding-top: 0px;
	height: 100%;
	display: flex;
	justify-content: space-between;

	.logout-text {
		position: absolute;
		bottom: 67px;
		right: 11px;
		color: black;
		z-index: 1000;
		border-radius: 50%;
		transition: 0.1s;
		opacity: 1;
		font-family: 'Roboto';
		cursor: pointer;
		padding: 4px;
		font-size: 12px;
	}
}

h2 {
	color: $main;
}

.logout-icon {
	position: absolute;
	bottom: 30px;
	right: 15px;
	width: 35px;
	height: 35px;
	border-radius: 50%;
	transition: 0.1s;
	cursor: pointer;
	padding: 4px;

	&:hover {
		background: white;
		opacity: 0.7;
		box-shadow: 0 2px 3px rgba(0, 0, 0, 0.137);
		transform: rotate(-25deg);
	}

	&:active {
		transform: rotate(25deg);
	}
}

@media only screen and (max-width: 1100px) {
	.hidden-circle {
		height: 1200px;
		width: 1200px;
		background: $accent;
		position: relative;
		top: -950px;
		left: -300px;
		z-index: -1;
		border-radius: 50%;
		opacity: 0;
		transition: 0.2s;
		transition-delay: 0.2s;
	}

	.menu {
		.information-area {
			height: 80%;
			width: 850px;
			padding-left: 80px;
			display: block;

			h1 {
				color: $main;
				font-size: 50px;
				width: 300px;
			}
		}
	}
}

@media only screen and (max-width: 835px) {
	.hidden-circle {
		height: 1200px;
		width: 1200px;
		// background: none;
		position: relative;
		top: -950px;
		left: -300px;
		z-index: -1;
		border-radius: 50%;
		opacity: 0;
		transition: 0.2s;
		transition-delay: 0.2s;
	}
	.menu {
		.information-area {
			height: 80%;
			width: 850px;
			padding-left: 80px;
			display: block;

			h1 {
				color: $main;
				font-size: 50px;
				width: 300px;
			}
		}
	}
}
