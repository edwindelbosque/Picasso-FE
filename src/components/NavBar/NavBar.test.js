import React from 'react';
import NavBar from './NavBar';
import { shallow } from 'enzyme';

describe('NavBar', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<NavBar />);
	});

	it('App should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
