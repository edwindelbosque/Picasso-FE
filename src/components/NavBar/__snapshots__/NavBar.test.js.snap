// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar App should match snapshot 1`] = `undefined`;
